static/extensions/MubiLop/penguingpt.js

```diff
@@ -0,0 +1,752 @@
+                })
+                .then(data => {
+                    return JSON.stringify(data);
+                })
+                .catch(error => {
+                    console.error("Error sending prompt to Moderation Api", error.message);
+                    return "Error: ", error.message;
+                });
+        }
+
+        checkApiUrl() {
+            // Send a simple GET request to the api_url			
+            return Scratch.fetch(api_url, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                        'Origin': 'https://gptcall.net/',
+                        'Referer': 'https://gptcall.net/'
+                    },
+                    body: JSON.stringify({
+                        model: "gpt-3.5-turbo",
+                        messages: [{
+                            role: "user",
+                            content: "Return nothing"
+                        }]
+                    }),
+                })
+                .then(response => {
+                    // Check if the response status code is in the 200 range (success)
+                    return response.status >= 200 && response.status < 300;
+                })
+                .catch(error => {
+                    // If there's an error, return false
+                    return false;
+                });
+        }
+
+        singlePrompt(args) {
+            const prompt = args.PROMPT;
+
+            return Scratch.fetch(api_url, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                        'Origin': 'https://gptcall.net/',
+                        'Referer': 'https://gptcall.net/'
+                    },
+                    body: JSON.stringify({
+                        model: this.model,
+                        messages: [{
+                            role: "user",
+                            content: prompt
+                        }]
+                    }),
+                })
+                .then(response => {
+                    if (!response.ok) {
+                        throw new Error(`Network response was not ok: ${response.status} ${response.statusText}`);
+                    }
+                    return response.json();
+                })
+                .then(data => {
+                    const botResponse = data.choices[0].message.content;
+                    return botResponse;
+                })
+                .catch(error => {
+                    console.error("Error sending prompt to GPT", error.message);
+                    return "Error: ", error.message;
+                });
+        }
+
+        generateImage(args) {
+            const prompt = args.PROMPT;
+            const requestedModel = args.MODEL
+
+            return Scratch.fetch("https://reverse.mubi.tech/image/generate", {
```

Comment:
Does not respect `api_url` variable, either disable the ability to change it or use the variable here